File: src/Domain/Library/Publisher.cs
using SharedKernel;

namespace Domain.Library;

public sealed class Publisher : Entity
{
    public Publisher()
    {
        Books = new List<Book>();
    }

    public int Id { get; set; }
    public string Name { get; set; } = string.Empty;
    public string ManagerName { get; set; } = string.Empty;
    public int PlaceId { get; set; }
    public string Address { get; set; } = string.Empty;
    public string Telephone { get; set; } = string.Empty;
    public string Website { get; set; } = string.Empty;
    public string Email { get; set; } = string.Empty;

    public ICollection<Book> Books { get; set; }
}

File: src/Web.UI/src/app/models/publisher.ts
export type Publisher = {
  id: number;
  name: string;
  managerName: string;
  telephone: string;
  address: string;
  cityId?: number;
  cityName?: string;
};

export type PublisherName = {
  id: number;
  title: string;
};

export type CreatePublisherRequest = {
  name: string;
  managerName: string;
  telephone: string;
  address: string;
  cityId: number;
};

export type UpdatePublisherRequest = CreatePublisherRequest & {
  id: number;
};


